src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -1061,7 +1062,215 @@ fromJSONInstDecl typeName =
 
 -- *** Generate `ToSchema` instance
 
-toSchemaInstanceDeclaration
+toSchemaInstanceDeclarationNew
+    :: MonadError CompileError m
+    => String
+    -- ^ Name of the message type to create an instance for
+    -> Maybe [HsName]
+    -- ^ Oneof constructors
+    -> [(String, DotProtoMessagePart)]
```

Comment:
Actually, looking at this more closely, I think the issue is that the change you made to the expected test output is incorrect.  For example, the `name` and `someid` fields of the `oneof` should not be marked required.  In protocol buffers, the fields of a `oneof` clause are like the alternative constructors of a sum type: protocol buffers requires at most one of them to be present, and definitely not more than one to be present.

However, we're using `swagger2`, which is based on an older version of the OpenAPI specification.  Specifically, the specification does not provide a way to specify that exactly one field out of many can be required.  Newer versions of the OpenAPI specification support this, but we're not using that, yet.  The best thing you can do for now is to make none of the fields of the `oneof` required.

That said, there is still a change you can make to avoid creating two separate `toSchemaInstanceDeclaration*` functions.  You can change this function argument type to:

```suggestion
    -> [(String, Bool)]
```

If you change `toSchemaInstanceDeclarationNew` to take a `Bool` here instead of a `DotProtoMessagePart` then you can use it in both places.  The idea is that the `Bool` represent whether or not the field is required.

In one place, you just use the output of the `isRequired` function to compute this `Bool` (like you already do, but now the call to `isRequired` will be in a different place).  In the other place (inside of `nestedOneOfDecls`), you can hard-code the `Bool` to `False` since none of the `oneof` fields are required.  Then you don't need to have `DotProtoMessagePart`s for the second place.